persistence/lock-keeper-sql/src/api.rs

```
@@ -0,0 +1,456 @@
+    ) -> Result<Option<User>, PostgresError> {
+        info!("Searching for user by ID");
+
+        let user_db: Option<UserDB> = sqlx::query_as!(
+            UserDB,
+            "SELECT account_id, user_id, account_name, storage_key, server_registration \
+            FROM Accounts \
+            WHERE user_id=$1",
+            user_id.as_ref()
+        )
+        .fetch_optional(&self.connection_pool)
+        .await?;
+
+        let user = user_db.map(User::try_from).transpose()?;
+        Ok(user)
+    }
+
+    #[instrument(skip_all, err(Debug), fields(user_id=?user_id))]
+    pub(crate) async fn delete_user(&self, user_id: &UserId) -> Result<(), PostgresError> {
+        info!("Deleting user.");
+
+        // Delete the entry and
+        let result = sqlx::query!(
+            r#"WITH deleted AS (DELETE FROM Accounts WHERE user_id=$1 RETURNING *) SELECT count(*) AS "count!" FROM deleted"#,
+            user_id.as_ref()
+        )
+        .fetch_one(&self.connection_pool)
+        .await?;
+
+        // No row was deleted.
+        match result.count {
+            0 => Err(PostgresError::NoEntry),
+            1 => Ok(()),
+            _ => Err(PostgresError::InvalidRowCountFound),
+        }
+    }
+
+    #[instrument(skip_all, err(Debug), fields(user_id=?user_id))]
+    pub(crate) async fn set_storage_key(
+        &self,
+        user_id: &UserId,
+        storage_key: Encrypted<StorageKey>,
+    ) -> Result<(), PostgresError> {
+        info!("Setting storage key for");
+
+        let storage_key = rmp_serde::to_vec(&storage_key)?;
+
+        let _ = sqlx::query!(
+            "UPDATE Accounts SET storage_key=$1 WHERE user_id=$2",
+            storage_key,
+            user_id.as_bytes(),
+        )
+        .execute(&self.connection_pool)
+        .await?;
+
+        Ok(())
+    }
+}
+
+/// Create a SQL query list of the form (val1, val2, ...). This function does
+/// not handle empty iterators. The iterator should have at least one element.
```

I think this should return an error if this is the case.